lib/logstash/inputs/cloudwatch_logs.rb

```diff
@@ -229,23 +307,56 @@ def parse_time(data)
 
   private
   def _sincedb_open
-    begin
-      File.open(@sincedb_path) do |db|
-        @logger.debug? && @logger.debug("_sincedb_open: reading from #{@sincedb_path}")
-        db.each do |line|
-          group, pos = line.split(" ", 2)
-          @logger.debug? && @logger.debug("_sincedb_open: setting #{group} to #{pos.to_i}")
-          @sincedb[group] = pos.to_i
+    unless @dynamodb_sincedb_table.nil?
+      #get the sincedb from ddb
+      begin
+        @logger.debug? && @logger.debug("_sincedb_open: reading from DynmanoDB: #{@dynamodb_sincedb_table}")
+        resp = @ddb.scan(
+          {
+            table_name: @dynamodb_sincedb_table,
+          }
+        )
+        resp.items.each do |item|
+          @logger.debug? && @logger.debug("_sincedb_open: setting #{item["GroupName"]} to #{item["StartPosition"]}")
+          @sincedb[item["GroupName"]] = item["StartPosition"].to_i
+        end
+      rescue
+        #problem opening DynamoDB
+        @logger.debug? && @logger.debug("_sincedb_open: error: #{@dynamodb_sincedb_table}: #{$!}")
+      end
+    else
+      begin
+        File.open(@sincedb_path) do |db|
+          @logger.debug? && @logger.debug("_sincedb_open: reading from #{@sincedb_path}")
+          db.each do |line|
+            group, pos = line.split(" ", 2)
+            @logger.debug? && @logger.debug("_sincedb_open: setting #{group} to #{pos.to_i}")
+            @sincedb[group] = pos.to_i
+          end
         end
+      rescue
+        #No existing sincedb to load
+        @logger.debug? && @logger.debug("_sincedb_open: error: #{@sincedb_path}: #{$!}")
       end
-    rescue
-      #No existing sincedb to load
-      @logger.debug? && @logger.debug("_sincedb_open: error: #{@sincedb_path}: #{$!}")
     end
   end # def _sincedb_open
 
+  private
+  def update_ddb_group(group)
+    resp = @ddb.put_item(
+      {
+        table_name: @dynamodb_sincedb_table,
+        item: {
+          "GroupName" => group,
+          "StartPosition" => "#{@sincedb[group]}"
+        }
+      }
+    )
+  end
```

Comment:
What would happen if we `errored` here due to something like dynamo throttles? Should special handling be created for that case?

Reply:
Yes, I should have put in some error checking here :)